be/src/olap/tablet_manager.cpp

```diff
@@ -232,6 +236,19 @@ OLAPStatus TabletManager::create_tablet(const TCreateTabletReq& request,
     LOG(INFO) << "begin to create tablet. tablet_id=" << tablet_id
               << ", schema_hash=" << schema_hash;
 
+    if (FilePathDesc::is_remote(request.storage_medium)) {
+        FilePathDesc path_desc;
+        path_desc.storage_medium = request.storage_medium;
+        path_desc.storage_name = request.storage_param.storage_name;
+        std::shared_ptr<Env> env = Env::get_env(path_desc);
+        if (env == nullptr) {
+            LOG(INFO) << "remote storage is not exist, create it. storage_name: " << request.storage_param.storage_name;
+            RETURN_WITH_WARN_IF_ERROR(Env::get_remote_mgr()->create_remote_storage(
```

Comment:
Remote Storage Object should be shared, do not create remote storage object for every tablet. For example, we may monitor the remote storage's performance, if there are too many object, it is too hard.

Reply:
Remote storage is shared in RemoteEnvMgr. Env::get_env() will get the shared_ptr from RemoteEnvMgr, path_desc will only take the storage_name and medium type.